### PR TITLE
cmake: Port PR25972 from the master branch

### PR DESCRIPTION
**Pull request by hebasto**
This PR ports https://github.com/bitcoin/bitcoin/pull/25972 after the recent sync/rebase [PR](https://github.com/hebasto/bitcoin/pull/186).

### CODE REVIEW

File: .github/workflows/cmake.yml [hunk 1]
```diff
@@ -286,12 +286,13 @@ jobs:
...
             packages: 'g++-mingw-w64-x86-64-posix'
             depends_options: ''
+            configure_options: '-DWERROR=ON -DCMAKE_CXX_FLAGS="-Wno-error=maybe-uninitialized"'
             exe_extension: '.exe'
           - name: 'MinGW-w64, debug'
             triplet: 'x86_64-w64-mingw32'
             packages: 'g++-mingw-w64-x86-64-posix'
             depends_options: 'DEBUG=1'
-            configure_options: '-DCMAKE_BUILD_TYPE=Debug'
+            configure_options: '-DCMAKE_BUILD_TYPE=Debug -DWERROR=ON -DCMAKE_CXX_FLAGS="-Wno-error=maybe-uninitialized"'
```
Comment (vasild) [inline .github/workflows/cmake.yml:295]: Are we not supposed to use the `APPEND_CXXFLAGS` for this?

**Review by vasild — APPROVED** [review]
ACK b748fbfa8118d30f0b2b2bde271a47de76460ec0

`-DCMAKE_CXX_FLAGS=...` would have to be adjusted after `APPEND_CXXFLAGS` is done. Maybe postpone this PR until then or drop the last commit from this PR.

**Comment by hebasto** [comment]
> `-DCMAKE_CXX_FLAGS=...` would have to be adjusted after `APPEND_CXXFLAGS` is done. Maybe postpone this PR until then or drop the last commit from this PR.

The last commit has been dropped.

**Review by vasild — APPROVED** [review]
ACK 1689b48e80c227390df6a0bc7f80843472fb7806

**Comment by fanquake** [comment]
I don't see how the last commit can be dropped, unless the CI here isn't equivalent to master? Otherwise it should now be failing with build errors?

**Comment by fanquake** [comment]
I guess I don't understand why compilers are being changed to avoid warnings then? Why not just turn off Werror if the point isn't to match master? Rather than swap to a compiler we don't use?

**Comment by hebasto** [comment]
> I guess I don't understand why compilers are being changed to avoid warnings then? Why not just turn off Werror if the point isn't to match master?

Yes. This can be done.

> Rather than swap to a compiler we don't use?

The number of developers, who uses Ubuntu 24.04, is expected to grow.

Anyway, all CI stuff has to be resolved in #142.

**Review by Theghost256 — APPROVED** [review]

**Pull request merged**